shell/browser/native_window_mac.h

```diff
@@ -214,9 +217,15 @@ class NativeWindowMac : public NativeWindow, public ui::NativeThemeObserver {
   bool zoom_to_page_width_ = false;
   bool fullscreen_window_title_ = false;
   bool resizable_ = true;
-  bool exiting_fullscreen_ = false;
   gfx::Point traffic_light_position_;
 
+  enum class FullScreenTransitionState { ENTERING, EXITING, NONE };
```

Comment:
It doesn't seem to me that there's any difference between `ENTERING` and `EXITING` in terms of behavior here. Can this be collapsed into two states?

Reply:
@nornagon we specifically need to check entering for redrawing of traffic lights, which is why i kept these separate